camel-idea-plugin/src/main/java/com/github/cameltooling/idea/service/QueryUtils.java

```diff
@@ -28,13 +28,22 @@ private QueryUtils() {
      * Validate if the query contain a known camel component
      */
     public static boolean isQueryContainingCamelComponent(Project project, String query) {
-        // is this a possible Camel endpoint uri which we know
-        if (query != null && !query.isEmpty()) {
-            String componentName = StringUtils.asComponentName(query);
-            if (componentName != null && project.getService(CamelCatalogService.class).get().findComponentNames().contains(componentName)) {
-                return true;
-            }
+        boolean isPossibleCamelEndpoint = query != null && !query.isEmpty();
+        if (!isPossibleCamelEndpoint) {
+            return false;
+        }
+
+        String componentName = StringUtils.asComponentName(query);
+        boolean isComponentNameValid = componentName != null;
+        if (!isComponentNameValid) {
+            return false;
+        }
+
+        boolean isComponentNameFound = project.getService(CamelCatalogService.class).get().findComponentNames().contains(componentName);
+        if (isComponentNameFound) {
+            return true;
+        } else {
+            return false;
```

Comment:
Not convinced by the utility of this change, it is really controversial, could you please revert?